[Keerthi G Kaarthik] iP

keerthigkaarthik opened the pull request:
Bob
> My best fren - Me

Bob helps you remember the things you need to do so you can focus on actually doing them! He is

- text-based
- based
- QUICK

You just need to

1. download
2. double click
3. add your tasks
4. profit!!

Bob is so kinda and willing to do all of this (and more in the future) for **free**

- [X] Multiple Tasks type
- [X] Deadlines
- [X] Mark as complete
- [ ] Sorting (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Eclipse-Dominator committed:
Bump gradle and lib version

build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

keerthigkaarthik committed:
Chang Duke to Bob

keerthigkaarthik committed:
Add echo

keerthigkaarthik committed:
Add add task

keerthigkaarthik committed:
Add done and undone task

keerthigkaarthik committed:
Task now abstract Add Deadline, Event, ToDo tasks

keerthigkaarthik committed:
Add minimal error

keerthigkaarthik committed:
Add delete task

keerthigkaarthik committed:
Improve format and personality

keerthigkaarthik committed:
Add try, catch and new exception

keerthigkaarthik committed:
Add auto test

keerthigkaarthik committed:
Add intro to auto test

keerthigkaarthik committed:
Add storage

keerthigkaarthik committed:
Add datetime

keerthigkaarthik committed:
Split classes for more OOP

keerthigkaarthik committed:
Add find

keerthigkaarthik committed:
Add JUnit test for Storage and Parser

keerthigkaarthik committed:
Add comments to comply with JavaDoc

keerthigkaarthik committed:
Add gradle and JUnit under Test directory

keerthigkaarthik committed:
Add gradle and JUnit under Test directory

keerthigkaarthik committed:
Merge branch 'master' of https://github.com/keerthigkaarthik/ip

* 'master' of https://github.com/keerthigkaarthik/ip:
  Add gradle and JUnit under Test directory
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version

keerthigkaarthik committed:
Revert "build.gradle: Prevent generating a second JAR file"

This reverts commit 4980f2897dff0738e08ced29f1855be425dd1ef3.

keerthigkaarthik committed:
Update Bob.java

keerthigkaarthik committed:
Merge pull request #1 from keerthigkaarthik/branch-A-CodingStandard

Update Bob.java

keerthigkaarthik committed:
Update UI.java

keerthigkaarthik committed:
Merge pull request #2 from keerthigkaarthik/branch-A-JavaDoc

Update UI.java

keerthigkaarthik committed:
Update Storage.java

keerthigkaarthik committed:
Merge pull request #3 from keerthigkaarthik/branch-Level-9

Update Storage.java

keerthigkaarthik committed:
Update Task.java

keerthigkaarthik committed:
Merge pull request #4 from keerthigkaarthik/branch-Level-8

Update Task.java

valhrd commented on src/main/java/Task/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+        int index = 1;
+        System.out.println();
+        for (Task task : this.tasks) {
+            System.out.printf("     %d. %s\n", index, task);
```
Nice use of format specifiers, I never thought of that actually.

valhrd commented on src/main/java/UI/Parser.java:
```diff
@@ -0,0 +1,142 @@
+    }
+
+    private static void handleDeadline(String arguments) throws BotException {
+        int byIndex = arguments.indexOf(" /by ");
+        if (byIndex == -1) throw new BotException("Please format your instructions correctly. E.g., deadline [task] /by [MMddyyyy HHmm]");
+        String taskDescription = arguments.substring(0, byIndex);
+        String deadline = arguments.substring(byIndex + 5);
```
Have you considered:
```suggestion
        String deadline = arguments.split(" /by ")[1].trim();
        if (deadline.isEmpty()) {
            throw new BotException("Please format your instructions correctly. E.g., deadline [task] /by [MMddyyyy HHmm]");
        }
```

keerthigkaarthik replied:
This would throw an invalid index error if there was no " /by " but I could just wrap it with a try block. Your suggestions looks a lot more concise thanks!

valhrd commented on src/main/java/UI/UI.java:
```diff
@@ -0,0 +1,68 @@
+     * Displays a separator line to the console.
+     */
+    public void showSeparator() {
+        System.out.println("--------------------------------------------------");
```
Can we try setting this as a final field?
```suggestion
        private static final DIVIDER = "--------------------------------------------------"
        // Code
        ...
        System.out.println(DIVIDER);
```

valhrd commented on src/main/java/UI/UI.java:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * Displays a separator line followed by a newline to the console.
+     */
+    public void showNewLineSeparator() {
+        System.out.println("--------------------------------------------------\n");
+    }
```
Can we try using a static final field?
```suggestion
    public void showNewLineSeparator() {
        System.out.println(DIVIDER + "\n");
    }
```

valhrd reviewed (commented):
Overall really well written code!

keerthigkaarthik committed:
Update src/main/java/UI/UI.java

Co-authored-by: Val <122347049+valhrd@users.noreply.github.com>

keerthigkaarthik committed:
Update src/main/java/UI/UI.java

Co-authored-by: Val <122347049+valhrd@users.noreply.github.com>

zaidansani commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,82 @@
+
+    private final String task;
+    public String type;
+    private boolean completed;
```
since this is a boolean variable, it would be better to name it with prefixes - in this case, perhaps `isCompleted`

zaidansani replied:
just noted your method is name `isCompleted`, perhaps that can be replaced with `checkIfCompleted`?

zaidansani commented on src/main/java/Task/Deadline.java:
```diff
@@ -0,0 +1,77 @@
+     *
+     * @return The formatted string for saving the deadline task to file.
+     */
+    @Override
```
there's a bit of repetition in the javadocs, perhaps to make it easier for you, if there are more tasks to be added, would be to utilise the `@inheritDocs` comment, and just add the specific difference in the overridden method after instead?

zaidansani replied:
specifically referring to this method since you have long comments for the abstract method and the overridden one as well, with repeated information

zaidansani commented on src/main/java/Task/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+     * @param term The term to search for within task descriptions.
+     * @return A TaskList of all tasks containing the term.
+     */
+    public TaskList tasksContainingTerm(String term) {
```
generally, since this is a method, it should be a verb - `getTasksContainingTerm` to indicate that it is a method instead of a variable

zaidansani commented on src/main/java/Task/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+     */
+    public void unmarkTask(int index) {
+        if (index >= 0 && index < tasks.size()) {
+            this.tasks.get(index).markAsUndone();
```
we can leave out the `this.` here to minimise noise, since there isn't another local `tasks` variable to confuse it with!

zaidansani commented on src/main/java/Bob/Bob.java:
```diff
@@ -0,0 +1,41 @@
+import UI.Parser;
+import UI.BotException;
+
+/**
```
your comments are excellent and descriptive in general! well done!

zaidansani commented on src/main/java/UI/Parser.java:
```diff
@@ -0,0 +1,142 @@
+package UI;
```
package names should generally be lowercase

zaidansani commented on src/main/java/UI/UI.java:
```diff
@@ -0,0 +1,71 @@
+ * Handles user interactions such as displaying messages and receiving input.
+ * Hello!
+ */
+public class UI {
```
this should probably be `Ui` instead, in accordance to convention

zaidansani commented on src/test/java/UI/ParserTest.java:
```diff
@@ -0,0 +1,56 @@
+    /**
+     * Sets up the environment before each test, initializing the TaskList with predefined tasks.
+     */
+    @BeforeEach
```
i like this approach to the tests!

zaidansani commented on src/main/java/Task/ToDo.java:
```diff
@@ -0,0 +1,43 @@
+     */
+    public static void load(String[] properties) {
+        ToDo newTodo = new ToDo(properties[2]);
+        if (Objects.equals(properties[1], "1")) {
```
out of curiosity, why use the objects.equals methods, and not just a `"1".equals(properties[1])`

zaidansani commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,82 @@
+    /**
+     * Marks the task as completed.
+     */
+    protected void markAsDone() {
```
since this is effectively a boolean method, it might be better to name this `setIsComplete` instead, based on coding convention!

zaidansani reviewed (commented):
hi keerthig, excellent work so far - i think some areas of improvement could be your naming (package names and class name conventions).

your comments are EXCELLENT - they're comprehensive, and informative!

skyl3-r commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+package Storage;
```
Package names should be in lower case in order to avoid conflict with classes or interfaces.

skyl3-r commented on src/main/java/Bob/Bob.java:
```diff
@@ -0,0 +1,41 @@
+package Bob;
```
It might make more sense to place the storage, task and ui packages as well as other bob related files in the bob package.

skyl3-r commented on src/main/java/UI/UI.java:
```diff
@@ -0,0 +1,71 @@
+     * Displays a separator line to the console.
+     */
+    public void showSeparator() {
+        private static final DIVIDER = "--------------------------------------------------"
```
Just curious, why is the constant DIVIDER defined in a non static method? What happens if showNewLineSeparator is run before showSeparator?

skyl3-r commented on src/main/java/UI/Parser.java:
```diff
@@ -0,0 +1,142 @@
+    private static void handleMarkUnmark(String command, String arguments) throws BotException {
+        if (arguments.isEmpty()) throw new BotException("Please provide a task number.");
+        int taskIndex = Integer.parseInt(arguments) - 1;
+        if (taskIndex < 0 || taskIndex >= TaskList.mainTaskList.getNumTasks()) throw new BotException("That task does not exist!");
```
I think the if statement should be formatted as
```suggestion
if (taskIndex < 0 || taskIndex >= TaskList.mainTaskList.getNumTasks()) {
    throw new BotException("That task does not exist!");
}
```
so that the line isn't too long and it is easier to read.

skyl3-r commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+import Task.Deadline;
+import Task.Event;
+import Task.TaskList;
+import Task.ToDo;
```
Good job on listing the imported classes explicitly rather than using *.

skyl3-r commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+/**
+ * The Storage class handles loading and saving data to a specified file path.
+ * It supports operations to clear, save, and load data.
+ * Hello!!
```
Good use of javadocs overall! Just not sure if the Hellos need to be there haha.

skyl3-r reviewed (commented):
Overall, I think you almost always stuck to style convention.  Good job!

keerthigkaarthik committed:
Delete src/test/java/UI directory

Resolve merge conflict

keerthigkaarthik committed:
Add GUI

keerthigkaarthik committed:
Merge pull request #7 from keerthigkaarthik/branch-Level-10

Add GUI

keerthigkaarthik committed:
Add remaining files

keerthigkaarthik committed:
Update main

keerthigkaarthik committed:
Improve code quality by refactoring and removing dead code

- Deleted unused and redundant code segments to enhance maintainability.
- Abstracted `showWelcome` method in the `MainWindow` class using SLAP
  to improve readability and modularity.
- Introduced enums for `TaskType` and `ReadBy` fields to replace
  hard-coded values, ensuring better type safety and readability.

  Both fields have fewer than four possible values, making enums an
  ideal choice.

keerthigkaarthik committed:
Add assert file path in Storage

keerthigkaarthik committed:
Merge pull request #8 from keerthigkaarthik/branch-A-Assertions

Add assert file path in Storage

keerthigkaarthik committed:
Merge pull request #9 from keerthigkaarthik/branch-A-CodeQuality

Improve code quality by refactoring and removing dead code

keerthigkaarthik committed:
Improve Chatbot GUI with Enhanced Styling

- Updated the chatbot GUI to enhance visual appeal
- Added CSS files for improved styling with colours, padding,
  and layout adjustments
- Replaced the plain white background with a more polished look

Improve user experience

keerthigkaarthik committed:
Add B-Reminders

keerthigkaarthik committed:
Add JUnit test for storage and improve code quality

keerthigkaarthik committed:
Add UI.png

keerthigkaarthik committed:
Update README.md

keerthigkaarthik committed:
Fix bye command

keerthigkaarthik committed:
Update Parser class methods handleBye and parse

keerthigkaarthik committed:
Rename UI.png to Ui.png

keerthigkaarthik committed:
Update build.gradle to include other OS and improve OOP in handleBye

keerthigkaarthik committed:
Merge branch 'master' of https://github.com/keerthigkaarthik/ip

* 'master' of https://github.com/keerthigkaarthik/ip:
  Rename UI.png to Ui.png

keerthigkaarthik committed:
Remove Duke README.md

keerthigkaarthik committed:
Fix handleClear and update README.md
